backend/services/workflows/app/processor/job.go

```diff
@@ -61,6 +63,18 @@ func (j JobProcessor) ProcessJSON(
 				len(key) > len(workflowInputVariable) {
 				key = key[len(workflowInputVariable):]
 				for _, param := range j.job.InputParameters {
+					// support for key === "device.json.tier"; accessing the input params
+					// objects by json identifiers. for instance if a whole device is passed
+					// as an input parameter then with `device.json.tier` you can refer
+					// to the Device.Tier field
+					if strings.HasPrefix(key, param.Name+".json") {
```

Comment:
I believe this format conflicts with the format for output variables
https://github.com/mendersoftware/mender-server/blob/e357e4983eaa04be17fb944b34286911234aa597/backend/services/workflows/app/processor/string.go#L35

Maybe something like `workflow.jsonInput.<inputName>.<jsonpath>` could work?